s3transfer/tasks.py

```diff
@@ -245,6 +157,97 @@ def _get_all_main_kwargs(self):
             kwargs[key] = result
         return kwargs
 
+    def _submit_task(self, executor, task):
+        future = executor.submit(task)
+        # Add this created future to the list of associated future just
+        # in case it is needed during cleanups.
+        self._transfer_coordinator.add_associated_future(future)
+        return future
+
+
+class SubmissionTask(Task):
+    """A base class for any submission task
+
+    Submission tasks are the top-level task used to submit a series of tasks
+    to execute a particular transfer.
+    """
+    def _main(self, transfer_future, **kwargs):
+        """
+        :type transfer_future: s3transfer.futures.TransferFuture
+        :param transfer_future: The transfer future associated with the
+            transfer request that tasks are being submitted for
+
+        :param kwargs: Any additional kwargs that you may want to pass
+            to the _submit() method
+        """
+        try:
+            # Before submitting any tasks, run all of the on_queued callbacks
+            on_queued_callbacks = get_callbacks(transfer_future, 'queued')
+            for on_queued_callback in on_queued_callbacks:
+                on_queued_callback()
+
+            # Once callbacks have been ran set the status to running.
+            self._transfer_coordinator.set_status_to_running()
```

Comment:
I talked to you about this and I looked back through the code and it looks like we do not even expose the ``status`` property on a ``Transferfuture`` (we only use the status internally) so a user could not even probe into what the ``status`` is of a future to report a discrepancy. What is your thoughts on tabling this to when we expose ``status`` on the future because we will have a better idea of what each state should mean given as we would only expose it if there is enough reason to make it a public feature?